psbt/psbt.go

```diff
@@ -0,0 +1,676 @@
+// Copyright (c) 2018 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+// An implementation of Partially Signed Bitcoin Transactions (PSBT).
+// The format is defined in BIP 174:
+// https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki
+package psbt
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// BIP-174 aka PSBT defined values
+
+// Key types are currently encoded with single bytes
+type PsbtKeyType = uint8
+
+const PsbtMagicLength = 5
+
+var (
+	PsbtMagic = [PsbtMagicLength]byte{0x70,
+		0x73, 0x62, 0x74, 0xff, // = "psbt" + 0xff sep
+	}
+)
+
+// The below are the known key types as per the BIP.
+// Unknown types may be accepted but will not be processed.
+const (
+
```

Comment:
Hmm, guessing these are the same type because they're used in the PsbtKeyValuePair{}.. I'd probably make them distinct types though, and use GlobalType, InputType, OutputType..